### PR TITLE
add quiz for for-loop

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: notes/03-data-processing/02b-quiz-for-loops.MD [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+The `s` String interpolator is missing which means `${width * height}` is not interpreted.
+
+If we add the String interpolator, then "area is 5" would be printed because the variables are evaluated as follows: 
+1. width = 1, height = 4 
+1. width = 2, height = 4 
+1. width = 3, height = 4
+1. width = 1, height = 5 
+1. width = 2, height = 5 
+1. width = 3, height = 5
```
Comment (wi101) [inline notes/03-data-processing/02b-quiz-for-loops.MD:43]: I think the order would be:
```suggestion
1. width = 1, height = 4 
2. width = 1, height = 5
3. width = 2, height = 4
4. width = 2, height = 5 
5. width = 3, height = 4 
6. width = 3, height = 5
```
in this case 5 is the 2nd printed line.

File: notes/03-data-processing/02b-quiz-for-loops.MD [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  user.email.foreach { email =>
+    println(email) 
+  }
+}
+```
+
+The first `foreach` iterates over all elements in the List.
+The second `foreach` is only executed if `email` is a `Some`. 
+You can think of an `Option` as a `List` with at most one element.
+
```
Comment (wi101) [inline notes/03-data-processing/02b-quiz-for-loops.MD:80]: could you specify the correct answer?

File: notes/03-data-processing/02b-quiz-for-loops.MD (hunk 1 above)
Comment (julien-truffaut) [reply notes/03-data-processing/02b-quiz-for-loops.MD:43]: You're right, thx

File: notes/03-data-processing/02b-quiz-for-loops.MD (hunk 2 above)
Comment (julien-truffaut) [reply notes/03-data-processing/02b-quiz-for-loops.MD:80]: done

**Review by wi101 — APPROVED** [review]
LGTM thanks!

**Pull request merged**